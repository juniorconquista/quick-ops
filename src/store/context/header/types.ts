export const ISLOADING = 'ISLOADING';
export const ERROR = 'ERROR';
export const RESET = 'RESET';
