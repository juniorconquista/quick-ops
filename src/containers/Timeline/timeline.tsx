import React, { memo } from 'react';
import { Box } from 'quarks-ui';

const Timeline: React.FC = () => <Box styling="row">Timeline</Box>;

export default memo(Timeline);
