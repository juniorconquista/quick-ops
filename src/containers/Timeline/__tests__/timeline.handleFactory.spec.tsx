import React from 'react';
import { cleanup, render, screen } from '@testing-library/react';

import timelineHandleFactory from '../timeline.handleFactory';
import { CARDTYPE } from '../timeline.types';

describe('Timeline handleFactory component', () => {
    afterEach(() => {
        jest.clearAllMocks();
        cleanup();
    });

    it('tests basic snapshot', () => {
        const Component = timelineHandleFactory(CARDTYPE.EXPENSE);
        const container = render(<Component />);
        expect(container.asFragment()).toMatchSnapshot();
    });

    it('should render the EXPENSE component', () => {
        const Component = timelineHandleFactory(CARDTYPE.EXPENSE);
        render(<Component />);
        expect(screen.queryByTestId('mocked-expense-component')).toBeTruthy();
    });

    it('should render the ACCOUNTABILITY_SUBMITTED component', () => {
        const Component = timelineHandleFactory(
            CARDTYPE.ACCOUNTABILITY_SUBMITTED,
        );
        render(<Component />);
        expect(
            screen.queryByTestId('mocked-accountability-component'),
        ).toBeTruthy();
    });

    it('should render the EVALUATION component', () => {
        const Component = timelineHandleFactory(CARDTYPE.EVALUATION);
        render(<Component />);
        expect(
            screen.queryByTestId('mocked-evaluation-component'),
        ).toBeTruthy();
    });
});
