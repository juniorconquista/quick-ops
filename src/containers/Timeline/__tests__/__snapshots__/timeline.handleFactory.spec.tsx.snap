// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Timeline handleFactory component tests basic snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-jrAGrp dMwTaT"
    data-testid="mocked-expense-component"
    width="100%"
  >
    <div
      class="sc-jrAGrp kUajXv"
      width="70%"
    >
      <div
        class="sc-jrAGrp kBxLey"
      >
        <div
          class="sc-kEjbxe MOhQA"
        >
          <svg>
            asterisk.svg
          </svg>
        </div>
        <h2
          class="sc-gsTCUz egnVlF"
          font-size="13px"
        >
          09/11/2020
        </h2>
      </div>
      <div
        class="sc-jrAGrp LSnzq"
        width="100%"
      >
        <div
          class="sc-jrAGrp cALEOf"
        >
          <h1
            class="sc-bdfBwQ cUfASn"
            font-size="[object Object]"
            font-weight="bold"
          >
            TIPO
          </h1>
          <h2
            class="sc-gsTCUz kPNLYl"
            color="text.contrast"
            font-size="[object Object]"
            font-weight="bold"
          />
          <p
            class="sc-iBPRYJ ldYJBv"
            font-size="14px"
          />
        </div>
        <div
          class="sc-jrAGrp FaEaB"
        >
          <h1
            class="sc-bdfBwQ cUfASn"
            font-size="[object Object]"
            font-weight="bold"
          >
            VALOR
          </h1>
          <h2
            class="sc-gsTCUz kPNLYl"
            color="text.contrast"
            font-size="[object Object]"
            font-weight="bold"
          >
            undefined undefined
          </h2>
          <p
            class="sc-iBPRYJ ldYJBv"
            font-size="14px"
          >
            Valor da nota: undefined undefined
          </p>
        </div>
        <div
          class="sc-jrAGrp eLbUUM"
        >
          <h1
            class="sc-bdfBwQ czsznJ"
            font-size="11px"
            font-weight="bold"
          >
            STATUS
          </h1>
        </div>
      </div>
    </div>
    <div
      class="sc-jrAGrp isxMhl"
    >
      <p
        class="sc-iBPRYJ ejhUdQ"
      >
         Ver nota fiscal
      </p>
    </div>
  </div>
</DocumentFragment>
`;
